DiscordBot.DataAccess/EventsSheetsService.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.Services;
+using Google.Apis.Sheets.v4;
+using Google.Apis.Sheets.v4.Data;
+using static Google.Apis.Sheets.v4.SpreadsheetsResource.ValuesResource;
+
+namespace DiscordBot.DataAccess
+{
+    public class EventsSheetsService
+    {
+        private static readonly string[] scopes = { SheetsService.Scope.Spreadsheets };
+        private static readonly string applicationName = "Softwire Discord Bot";
+        private static readonly string spreadsheetId = "";
+
+        private SheetsService sheetsService;
+
+        private static async Task<ServiceAccountCredential> GetCredentialsAsync(string path = "credentials.json")
+        {
+            await using var stream =
+                new FileStream(path, FileMode.Open, FileAccess.Read);
+
+            var googleCredential = GoogleCredential.FromStreamAsync(stream, CancellationToken.None);
+
+            return (await googleCredential)
+                .CreateScoped(scopes)
+                .UnderlyingCredential as ServiceAccountCredential;
+        }
+
+        public EventsSheetsService() { }
+
+        public async Task StartService()
+        {
+            var credential = await GetCredentialsAsync();
+
+            // Create Google Sheets API service.
+            sheetsService = new SheetsService(new BaseClientService.Initializer
+            {
+                HttpClientInitializer = credential,
+                ApplicationName = applicationName
+            });
+        }
+
+        public async Task ReadColumns()
+        {
+            var range = "A2:B27";
+            var request =
+                sheetsService.Spreadsheets.Values.Get(spreadsheetId, range);
+
+            var response = await request.ExecuteAsync();
+            var values = response.Values;
+            if (values != null && values.Count > 0)
+            {
+                Console.WriteLine("Alpha, Numeric");
```

Comment:
Are the console logs a way to test that it's working?

Reply:
Yep, this is just a demo of getting Google Sheets to talk to the C# code. This will all be removed and replaced eventually by more useful methods implementing methods that Frank is finishing off the interfaces for 🙂